src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java

```diff
@@ -0,0 +1,116 @@
+
+    @ArgumentCollection
+    private final RequiredReadInputArgumentCollection readArguments = new RequiredReadInputArgumentCollection();
+
+    @ArgumentCollection
+    private final IntervalArgumentCollection intervalArgumentCollection = new OptionalIntervalArgumentCollection();
+
+    @ArgumentCollection
+    private final ReferenceInputArgumentCollection referenceArguments = new OptionalReferenceInputArgumentCollection();
+
+    @Argument(doc = "the known variants", shortName = "knownSites", fullName = "knownSites", optional = false)
+    private List<String> knownVariants;
+
+    @Argument(doc = "Path to save the final recalibration tables to.",
+              shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, optional = false)
+    private String outputTablesPath = null;
+
+    @Override
+    protected void runPipeline( JavaSparkContext ctx ) {
+        if ( readArguments.getReadFilesNames().size() != 1 ) {
+            throw new UserException("Sorry, we only support a single reads input for now.");
+        }
+        final String bam = readArguments.getReadFilesNames().get(0);
+
+        ReadsSparkSource readSource = new ReadsSparkSource(ctx);
+        SAMFileHeader readsHeader = ReadsSparkSource.getHeader(ctx, bam);
+        final List<SimpleInterval> intervals = intervalArgumentCollection.intervalsSpecified() ? intervalArgumentCollection.getIntervals(readsHeader.getSequenceDictionary())
+                : IntervalUtils.getAllIntervalsForReference(readsHeader.getSequenceDictionary());
+        JavaRDD<GATKRead> initialReads = readSource.getParallelReads(bam, intervals);
+
+        VariantsSparkSource variantsSparkSource = new VariantsSparkSource(ctx);
+        if ( knownVariants.size() > 1 ) {
+            throw new GATKException("Cannot currently handle more than one known sites file, " +
+                    "as getParallelVariants(List) is broken");
+        }
+        JavaRDD<Variant> bqsrKnownVariants = variantsSparkSource.getParallelVariants(knownVariants.get(0));
+
+        GCSOptions options = BucketUtils.getAuthenticatedGCSOptions(apiKey);
+        final String referenceURL = referenceArguments.getReferenceFileName();
+        final ReferenceDataflowSource referenceDataflowSource = new ReferenceDataflowSource(options, referenceURL, BaseRecalibratorDataflow.BQSR_REFERENCE_WINDOW_FUNCTION);
+        checkSequenceDictionaries(referenceDataflowSource.getReferenceSequenceDictionary(readsHeader.getSequenceDictionary()), readsHeader.getSequenceDictionary());
+
+        // TODO: Look into broadcasting the reference to all of the workers. This would make AddContextDataToReadSpark
+        // TODO: and ApplyBQSRStub simpler (#855).
+        JavaPairRDD<GATKRead, ReadContextData> rddReadContext = AddContextDataToReadSpark.add(initialReads, referenceDataflowSource, bqsrKnownVariants);
```

Comment:
Decided to modify `AddContextDataToReadSpark` itself to filter out unmapped reads, since it can't currently handle them.